diff --git a/ReactAndroid/src/main/java/com/facebook/react/fabric/mounting/SurfaceMountingManager.kt b/ReactAndroid/src/main/java/com/facebook/react/fabric/mounting/SurfaceMountingManager.kt
index 9224e19ddf3b1793162ff2ce86f950eff70fb06a..19feb2d12a9396e9c1e1d2e04df25bf67ffbfea3 100644
--- a/ReactAndroid/src/main/java/com/facebook/react/fabric/mounting/SurfaceMountingManager.kt
+++ b/ReactAndroid/src/main/java/com/facebook/react/fabric/mounting/SurfaceMountingManager.kt
@@ -1191,6 +1191,10 @@ internal constructor(
     ) {
       for ((propKey, propValue) in patchMap) {
         if (outputReadableMap.hasKey(propKey)) {
+          if (outputReadableMap.getType(propKey) == ReadableType.Null) {
+            continue
+          }
+
           if (propKey == PROP_TRANSFORM) {
             assert(outputReadableMap.getType(propKey) == ReadableType.Array && propValue is List<*>)
             val array = WritableNativeArray()
